bundle/edu.gemini.spModel.core/src/main/scala/edu/gemini/spModel/core/Target.scala

```diff
@@ -48,6 +48,23 @@ object Target {
       PLens.nil.run
     ))
 
+  val pm: Target @?> ProperMotion =
+    PLens(_.fold(
+      PLens.nil.run,
+      SiderealTarget.pm.run,
+      PLens.nil.run,
+      PLens.nil.run,
+      PLens.nil.run
+    ))
```

Comment:
A partial lens means it is not defined for all inputs, so the logical signature is `A => Option[B]`. In this case from a `Target` we want to get/set the `ProperMotion` but that is only valid for certain target types, specifically `SiderealTarget`.

I'm not totally familiar with the fold syntax but my understanding is that depending on the exact subtype it will run different `PLens`, being the only non-nil the one from `SiderealTarget`